### PR TITLE
Remove Trailers

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
Headers will be removed next, this is only part 1.

### CODE REVIEW

File: core/src/main/java/io/grpc/Metadata.java [hunk 1]
```diff
@@ -373,34 +378,6 @@ public String toString() {
   }
 
   /**
-   * Concrete instance for metadata attached to the end of the call. Only provided by
-   * servers.
-   */
-  public static class Trailers extends Metadata {
```
Comment (ejona86) [inline core/src/main/java/io/grpc/Metadata.java]: Could we leave this class present but marked as @ Deprecated?

Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/Metadata.java]: Like just this class, keep the rest of the PR?

Comment (ejona86) [inline core/src/main/java/io/grpc/Metadata.java]: Yeah. We could remove the byte[] constructor and the toString() without issue, but the public constructor is the main thing that existing code would be using.

Comment (carl-mastrangelo) [inline core/src/main/java/io/grpc/Metadata.java]: Done.

**Comment by ejona86** [comment]
@carl-mastrangelo LGTM

**Pull request merged**